### ISSUE TITLE
random_split is not fixed by seed

### ISSUE THREAD

**Issue by qqhann** [pytorch/pytorch#30364]
## ❓ Questions and Help

Even if I set seed, random_split is not returning same result from time to time.
Is this a bug or am I understanding random_split wrongly?

## To Reproduce

```
In [1]: import torch                                                                                                                                                                              

In [2]: import random                                                                                                                                                                             

In [3]: import numpy as np                                                                                                                                                                        

In [4]: SEED = 0 
   ...: random.seed(SEED) 
   ...: np.random.seed(SEED) 
   ...: torch.manual_seed(SEED) 
   ...: torch.backends.cudnn.deterministic = True 
   ...: torch.backends.cudnn.benchmark = False                                                                                                                                                    

In [5]: torch.__version__                                                                                                                                                                         
Out[5]: '1.3.0'

In [6]: sorted(torch.utils.data.random_split([1,2,3,4,5,6,7,8,9,0], [8,2])[0])                                                                                                                    
Out[6]: [0, 1, 2, 4, 5, 6, 8, 9]

In [7]: sorted(torch.utils.data.random_split([1,2,3,4,5,6,7,8,9,0], [8,2])[0])                                                                                                                    
Out[7]: [0, 1, 3, 4, 5, 7, 8, 9]
```

## Expected behavior

It should split data randomly the same way every time. Like:
```
In [6]: sorted(torch.utils.data.random_split([1,2,3,4,5,6,7,8,9,0], [8,2])[0])                                                                                                                    
Out[6]: [0, 1, 2, 4, 5, 6, 8, 9]

In [7]: sorted(torch.utils.data.random_split([1,2,3,4,5,6,7,8,9,0], [8,2])[0])                                                                                                                    
Out[7]: [0, 1, 2, 4, 5, 6, 8, 9]
```

## Environment

```
Collecting environment information...
PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
Nvidia driver version: 430.50
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.17.3
[pip] torch==1.3.0
[conda] torch
```

 - PyTorch Version (e.g., 1.0): 1.3.0
 - OS (e.g., Linux): Ubuntu 18.04.3 LTS
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7.4
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
 - Any other relevant information:

**Comment by qqhann** [comment]
Sorry, I misunderstood 

```In [6]: for _ in range(3): 
   ...:     SEED = 0 
   ...:     random.seed(SEED) 
   ...:     np.random.seed(SEED) 
   ...:     torch.manual_seed(SEED) 
   ...:     torch.backends.cudnn.deterministic = True 
   ...:     torch.backends.cudnn.benchmark = False 
   ...:     print(sorted(torch.utils.data.random_split(list(range(10)), [7,3])[0])) 
   ...:                                                                                                                                                                                           
[0, 1, 3, 4, 5, 7, 9]
[0, 1, 3, 4, 5, 7, 9]
[0, 1, 3, 4, 5, 7, 9]```
